jquery.flexslider.js

```diff
@@ -833,9 +856,43 @@
                        (reverse && slider.animatingTo === slider.last) ? 0 :
                        (reverse) ? slider.limit - (((slider.itemW + slider.vars.itemMargin) * slider.move) * slider.animatingTo) :
                        (slider.animatingTo === slider.last) ? slider.limit : posCheck;
+              } else if (variableImageWidth) {
+                var startWidth = 0,
+                  endWidth = 0,
+                  offsetWidth = 0,
+                  start = (reverse) ? ((slider.count - 1) - 0 + slider.cloneOffset) : (0 + slider.cloneOffset),
+                  offset = (reverse) ? ((slider.count - 1) - slider.currentSlide + slider.cloneOffset) : (slider.currentSlide + slider.cloneOffset),
+                  current = offset;
+
+                if(slider.animatingTo != slider.currentSlide) {
+                  offset = (reverse) ? ((slider.count - 1) - slider.animatingTo + slider.cloneOffset) : (slider.animatingTo + slider.cloneOffset);
+                }
+
+                if (slider.currentSlide === 0 && slider.animatingTo === slider.count - 1 && slider.vars.animationLoop && slider.direction !== "next") {
+                  offset = (reverse) ? slider.count + slider.cloneOffset : 0;
+                } else if (slider.currentSlide === slider.last && slider.animatingTo === 0 && slider.vars.animationLoop && slider.direction !== "prev") {
+                  offset = (reverse) ? 0 : slider.count + 1;
+                }
+
+                slider.newSlides.each(function(index) {
+                  var imgWidth = $(this).children('img').first().width();
+                  if(imgWidth === 0 || imgWidth === null) imgWidth = slider.computedW;
+                  if(index == current) slider.currentWidth = imgWidth;
+                  if(index < start) startWidth += imgWidth;
+                  if(index < offset) offsetWidth += imgWidth;
```

Comment:
Here you should additionally sum slider.vars.itemMargin to get itemMargin working correctly